src/project-css/app/Http/Controllers/CollectionController.php

```diff
@@ -51,36 +54,94 @@ public function index() {
     return view('admin/collection')->with('collcntNms', $collcntNms);
   }
 
+  /**
+  * Returns view for collection
+  *
+  * @return view
+  */    
+  public function show($colID) {
+    // find the collection
+    $thisClctn = Collection::findorFail($colID);  
+
+    // Get all the tables of this collection
+    $tbls = $thisClctn->tables()->get();
+
+    // redirect to show page
+    return view('collection/show')->with('tbls', $tbls)
+                                  ->with('colID', $colID)
+                                  ->with('clctnName', $thisClctn->clctnName);
+  }   
+  
+  /**
+  * Returns view for collection
+  *
+  * @return view
+  */    
+  public function upload($colID) {
+    // find the collection
+    $thisClctn = Collection::findorFail($colID);  
+
+    // redirect to show page
+    return view('collection/upload')->with('colID', $colID)
```

Comment:
The comment says returns a view, but in the code it shows that it is calling a view then returning a redirect.... this is a bit conspicous.